### PR TITLE
Fixed GSON parsing error when using configuration string

### PR DESCRIPTION
**Pull request by MEseifan**
Fix for [CDAP-16211](https://issues.cask.co/browse/CDAP-16211) was incomplete and it would result in a GSON error when attempting to deploy using a json string in the `configuration` field

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java
Comment (yaojiefeng) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java]: nit - I dont think we need this

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/AppFabricTestBase.java
Comment (yaojiefeng) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/AppFabricTestBase.java:533]: This might break unit test as some test might depend on the config, is this required for your unit test? If not lets not make this change

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java
Comment (yaojiefeng) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java:128]: just want to check - this test will fail without the fix right?

**Review by yaojiefeng — COMMENTED** [review]
Please also make sure you test it on a sandbox before merging

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java
Comment (MEseifan) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java]: Removed

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/AppFabricTestBase.java
Comment (MEseifan) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/AppFabricTestBase.java:533]: This is needed for my test since I need to test with a configuration of type `String` which doesnt work with `? extends Config`. I'm running all other tests that use this base to verify it didnt break them

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java
Comment (MEseifan) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java:128]: yes I removed the change and ran this test and it failed

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/AppFabricTestBase.java
Comment (yaojiefeng) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/AppFabricTestBase.java:533]: ok, let's just ensure the bamboo build passes.

**Review by yaojiefeng — APPROVED** [review]
LGTM assuming sandbox test and bamboo build passes

**Comment by MEseifan** [comment]
Manually tested in sandbox. Testing process: 
- Created sample pipeline and deployed it 
- Called `GET` on `/namespaces/default/apps/samplePipeline`
- Copied app spec that was returned and pasted it into the body of a `PUT` call to `/namespaces/default/apps/newPipeline`
- Verified that the new pipeline was deployed with the correct configuration

**Comment by MEseifan** [comment]
Bamboo build has also passed https://builds.cask.co/browse/CDAP-DUT7194-1

**Pull request merged**
